apiserver/common/secrets/secrets.go

```diff
@@ -535,3 +538,102 @@ func GetSecretMetadata(
 	}
 	return result, nil
 }
+
+// RemoveSecrets removes the specified secrets.
+// If removeFromBackend is false, the secrets are only removed from the state and
+// the caller must have permission to manage the secret(secret owners remove secrets from the backend on uniter side).
+// If removeFromBackend is true, the secrets are removed from the state and
+// backend and the caller must have model admin access.
+func RemoveSecrets(
+	secretsState SecretsMetaState,
+	removeState SecretsRemoveState,
+	secretsConsumer SecretsConsumer,
+	adminConfigGetter BackendAdminConfigGetter,
+	modelTag names.ModelTag, authorizer facade.Authorizer, authTag names.Tag, leadershipChecker leadership.Checker,
+	args params.DeleteSecretArgs, removeFromBackend bool,
+) (params.ErrorResults, error) {
```

Comment:
This is all a bit complicated and if then else rules for a single function. Having to invent a always false leadership checker and the different behaviours for unit agent vs client - we should just keep two separate functions, one for agent and one for client.